ethpm_types/utils.py

```diff
@@ -1,20 +1,75 @@
-from hashlib import md5
+from enum import Enum
+from hashlib import md5, sha3_256, sha256
 
 
-def compute_checksum(source: bytes, algorithm: str = "md5") -> str:
-    if algorithm == "md5":
-        hasher = md5
-    else:
-        raise Exception("Unknown algorithm")
+class Algorithm(str, Enum):
+    MD5 = "md5"
+    SHA3 = "sha3"
+    SHA256 = "sha256"
 
-    return hasher(source).hexdigest()
 
+def is_valid_hex(data: str) -> bool:
+    if not data.startswith("0x"):
+        return False
 
-def is_valid_hash(data: str) -> bool:
-    if set(data.lower()) > set("1234567890abcdef"):
+    if set(data[2:].lower()) > set("1234567890abcdef"):
         return False
 
     if len(data) % 2 != 0:
         return False
 
     return True
+
+
+class Hex(str):
+    """A hex string value, typically from a hash."""
+
+    @classmethod
+    def __modify_schema__(cls, field_schema):
+        field_schema.update(
+            pattern="^0x([0-9a-f][0-9a-f])*$",
+            examples=[
+                "0x",  # empty bytes
+                "0xd4",
+                "0xd4e5",
+                "0xd4e56740",
+                "0xd4e56740f876aef8",
+                "0xd4e56740f876aef8c010b86a40d5f567",
+                "0xd4e56740f876aef8c010b86a40d5f56745a118d0906a34e69aec8c0db1cb8fa3",
+            ],
+        )
+
+    @classmethod
+    def __get_validators__(cls):
+        yield cls.validate_hex
+
+    @classmethod
+    def validate_hex(cls, data: str) -> str:
+        if not is_valid_hex(data):
+            raise ValueError("Invalid Hex Value")
+
+        return data
+
+    @classmethod
+    def from_bytes(cls, data: bytes) -> "Hex":
+        return cls("0x" + data.hex())
+
+    def to_bytes(self) -> bytes:
+        return bytes.fromhex(self[2:])
+
+
+def compute_checksum(content: bytes, algorithm: Algorithm = Algorithm.MD5) -> Hex:
```

Comment:
Does this need a doc string?

Reply:
idk maybe. it is `utils` lol

Reply:
I don't see a problem with exposing it either lol